misc/inmanta-workon-register.sh

```diff
@@ -79,6 +79,124 @@ EOF
     __inmanta_workon_activate "$inmanta_env" "$env_id" "$envs_dir"
 }
 
+function __store_old_pip_config {
+    # Store pip configuration to be later restored in __restore_old_pip_config during deactivation
+    if [ -n "${PIP_PRE:-}" ] ; then
+        _OLD_INMANTA_CONFIG_PIP_PRE="${PIP_PRE:-}"
+    fi
+
+    if [ -n "${PIP_INDEX_URL:-}" ] ; then
+        _OLD_INMANTA_CONFIG_PIP_INDEX_URL="${PIP_INDEX_URL:-}"
+    fi
+
+    if [ -n "${PIP_EXTRA_INDEX_URL:-}" ] ; then
+        _OLD_INMANTA_CONFIG_PIP_EXTRA_INDEX_URL="${PIP_EXTRA_INDEX_URL:-}"
+    fi
+
+    if [ -n "${PIP_CONFIG_FILE:-}" ] ; then
+        _OLD_INMANTA_CONFIG_PIP_CONFIG_FILE="${PIP_CONFIG_FILE:-}"
+    fi
```

Comment:
I've been trying to find a way to factorize these code duplications into a function (and similarly for `__restore_old_pip_config` ) but I didn't manage to. Is there a clean/safe/obvious way to do this ?

Reply:
eg replace all these with 
```sh
...
__store PIP_EXTRA_INDEX_URL
__store PIP_INDEX_URL
...
```
And define a `__store` function implementing the logic

Reply:
I'm not sure either, so I would just leave it as is. I might think differently if it were all over the place, but it's all nicely centralized now.